databricks/koalas/tests/test_groupby.py

```diff
@@ -852,6 +852,62 @@ def test_rank(self):
             pdf.groupby([("x", "a"), ("x", "b")]).rank().sort_index(),
         )
 
+    def test_cumcount(self):
```

Comment:
Could we also have a tests for `ascending=False` ?

You could just use the below

```python
    def test_cumcount(self):
        pdf = pd.DataFrame(
            {
                "a": [1, 2, 3, 4, 5, 6] * 3,
                "b": [1, 1, 2, 3, 5, 8] * 3,
                "c": [1, 4, 9, 16, 25, 36] * 3,
            },
            index=np.random.rand(6 * 3),
        )
        kdf = ks.from_pandas(pdf)

        self.assert_eq(
            kdf.groupby("b").cumcount().sort_index(), pdf.groupby("b").cumcount().sort_index()
        )
        self.assert_eq(
            kdf.groupby(["a", "b"]).cumcount().sort_index(),
            pdf.groupby(["a", "b"]).cumcount().sort_index(),
        )
        self.assert_eq(
            kdf.groupby(["b"])["a"].cumcount().sort_index(),
            pdf.groupby(["b"])["a"].cumcount().sort_index(),
            almost=True,
        )
        self.assert_eq(
            kdf.groupby(["b"])[["a", "c"]].cumcount().sort_index(),
            pdf.groupby(["b"])[["a", "c"]].cumcount().sort_index(),
            almost=True,
        )
        self.assert_eq(
            kdf.groupby(kdf.b // 5).cumcount().sort_index(),
            pdf.groupby(pdf.b // 5).cumcount().sort_index(),
            almost=True,
        )
        self.assert_eq(
            kdf.groupby(kdf.b // 5)["a"].cumcount().sort_index(),
            pdf.groupby(pdf.b // 5)["a"].cumcount().sort_index(),
            almost=True,
        )
        self.assert_eq(
            kdf.groupby("b").cumcount().sum(), pdf.groupby("b").cumcount().sum(),
        )

        # ascending = False
        self.assert_eq(
            kdf.groupby("b").cumcount(ascending=False).sort_index(), pdf.groupby("b").cumcount(ascending=False).sort_index()
        )
        self.assert_eq(
            kdf.groupby(["a", "b"]).cumcount(ascending=False).sort_index(),
            pdf.groupby(["a", "b"]).cumcount(ascending=False).sort_index(),
        )
        self.assert_eq(
            kdf.groupby(["b"])["a"].cumcount(ascending=False).sort_index(),
            pdf.groupby(["b"])["a"].cumcount(ascending=False).sort_index(),
            almost=True,
        )
        self.assert_eq(
            kdf.groupby(["b"])[["a", "c"]].cumcount(ascending=False).sort_index(),
            pdf.groupby(["b"])[["a", "c"]].cumcount(ascending=False).sort_index(),
            almost=True,
        )
        self.assert_eq(
            kdf.groupby(kdf.b // 5).cumcount(ascending=False).sort_index(),
            pdf.groupby(pdf.b // 5).cumcount(ascending=False).sort_index(),
            almost=True,
        )
        self.assert_eq(
            kdf.groupby(kdf.b // 5)["a"].cumcount(ascending=False).sort_index(),
            pdf.groupby(pdf.b // 5)["a"].cumcount(ascending=False).sort_index(),
            almost=True,
        )
        self.assert_eq(
            kdf.groupby("b").cumcount(ascending=False).sum(), pdf.groupby("b").cumcount(ascending=False).sum(),
        )

        # multi-index columns
        columns = pd.MultiIndex.from_tuples([("x", "a"), ("x", "b"), ("y", "c")])
        pdf.columns = columns
        kdf.columns = columns

        self.assert_eq(
            kdf.groupby(("x", "b")).cumcount().sort_index(),
            pdf.groupby(("x", "b")).cumcount().sort_index(),
        )
        self.assert_eq(
            kdf.groupby([("x", "a"), ("x", "b")]).cumcount().sort_index(),
            pdf.groupby([("x", "a"), ("x", "b")]).cumcount().sort_index(),
        )

        # ascending = False
        self.assert_eq(
            kdf.groupby(("x", "b")).cumcount(ascending=False).sort_index(),
            pdf.groupby(("x", "b")).cumcount(ascending=False).sort_index(),
        )
        self.assert_eq(
            kdf.groupby([("x", "a"), ("x", "b")]).cumcount(ascending=False).sort_index(),
            pdf.groupby([("x", "a"), ("x", "b")]).cumcount(ascending=False).sort_index(),
        )
```

Reply:
Sounds good, but shall we use a loop for ascending?

```py
for ascending in [True, False]:
    ...
```

Reply:
Looks better 👍

Reply:
And maybe we could test for `[True, False, 0, 1, None]` - Not sure maybe It's too much though -
I fixed the example code in the above comment.